pd/src/state/writer.rs

```diff
@@ -238,6 +238,36 @@ impl Writer {
         .execute(&mut dbtx)
         .await?;
 
+        // Drop quarantined notes associated with a validator slashed in this block
+        for note_commitment in block.reverting_notes {
+            query!(
+                "DELETE FROM quarantined_notes WHERE note_commitment = $1",
+                &<[u8; 32]>::from(note_commitment)[..]
+            )
+            .execute(&mut dbtx)
+            .await?;
+        }
+
+        // Drop quarantined nullifiers from the main nullifier set if they were associated with a
+        // validator slashed in this block (thus reverting their spend)
+        for nullifier in block.reverting_nullifiers {
+            // Forget about this nullifier, making the associated note spendable again
+            query!(
+                "DELETE FROM nullifiers WHERE nullifier = $1",
```

Comment:
This is very dangerous; a bug that introduced a nullifier into `reverting_nullifiers` would enable a double-spend. Please check my work!

Reply:
I'm not seeing where `reverting_nullifiers` has any nullifiers added to it

Reply:
LGTM